android/sdl_android/src/main/java/com/smartdevicelink/util/SdlAppInfo.java

```diff
@@ -153,12 +153,12 @@ public int compare(SdlAppInfo one, SdlAppInfo two) {
                     int versionCompare =  two.routerServiceVersion  - one.routerServiceVersion;
 
                     if(versionCompare == 0){ //Versions are equal so lets use the one that has been updated most recently
-                        int updateTime =  (int)(two.lastUpdateTime - one.lastUpdateTime);
+                        long updateTime =  two.lastUpdateTime - one.lastUpdateTime;
                         if(updateTime == 0){
                             //This is arbitrary, but we want to ensure all lists are sorted in the same order
                             return  one.routerServiceComponentName.getPackageName().compareTo(two.routerServiceComponentName.getPackageName());
                         }else{
-                            return updateTime;
+                            return (updateTime < 0 ? -1 : 1);
                         }
                     }else{
                         return versionCompare;
```

Comment:
```suggestion
                        return (versionCompare < 0 ? -1 : 1);
```

Reply:
Since we're updating the other return statement, it would make sense to do it here too.